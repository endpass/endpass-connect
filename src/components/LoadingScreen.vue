<template>
  <div class="loading-screen">
    <div class="loading-screen__spinner">
      <Spinner />
    </div>
  </div>  
</template>

<script>
  import Spinner from '@/components/Spinner'

  export default {
    name: 'LoadingScreen',

    components: {
      Spinner,
    },
  };
</script>

<style lang="postcss">
.loading-screen {
  position: absolute; 
  display: flex;
  align-items: center;
  justify-content: center;
  top: 0;
  left: 0;
  z-index: 2;
  width: 100%;
  height: 100%;
}

.loading-screen__spinner {
  flex: 0 0 auto;
}
</style>