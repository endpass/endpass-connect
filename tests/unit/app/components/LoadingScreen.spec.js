import { shallowMount } from '@vue/test-utils';
import LoadingScreen from '@/components/LoadingScreen.vue';

describe('LoadingScreen', () => {
  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(LoadingScreen);
    });

    it('should correctly render LoadingScreen component', () => {
      expect(wrapper.name()).toBe('LoadingScreen');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
