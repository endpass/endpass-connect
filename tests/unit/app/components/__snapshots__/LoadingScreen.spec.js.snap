// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen render should correctly render LoadingScreen component 1`] = `
<div class="loading-screen">
  <div class="loading-screen__spinner">
    <spinner-stub></spinner-stub>
  </div>
</div>
`;
